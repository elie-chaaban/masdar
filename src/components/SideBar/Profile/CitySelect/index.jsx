import React from 'react';
import {Field} from 'formik';
import PropTypes from 'prop-types';

const CitySelect = ({name, enabled, cities, error, touched}) => {
  return (
    <div className="mt-2 col-6">
      <p>City</p>
      <Field name={name}>
        {({
          field // { name, value, onChange, onBlur }
        }) => (
          <>
            <select disabled={!enabled} {...field} className={`select-city ${enabled ? 'active' : ''}`}>
              <option value="">select city</option>
              {cities.map(({id, name}) => (
                <option key={id} value={id}>
                  {name}
                </option>
              ))}
            </select>
            {error && touched ? <div className="text-danger">{error}</div> : null}
          </>
        )}
      </Field>
    </div>
  );
};
CitySelect.protoTypes = {
  name: PropTypes.string.isRequired,
  enabled: PropTypes.bool.isRequired,
  cities: PropTypes.arrayOf(PropTypes.shape({id: PropTypes.string.isRequired, name: PropTypes.string.isRequired})),
  error: PropTypes.string,
  touched: PropTypes.bool.isRequired
};
export default CitySelect;
