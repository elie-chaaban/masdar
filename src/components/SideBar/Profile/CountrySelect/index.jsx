import React from 'react';
import {Field} from 'formik';
import PropTypes from 'prop-types';

const CountrySelect = ({name, enabled, countries, error, touched, onChange}) => {
  return (
    <div className="mt-2 col-6">
      <p>Country</p>
      <Field name={name}>
        {({
          field // { name, value, onChange, onBlur }
        }) => (
          <>
            <select
              disabled={!enabled}
              {...field}
              className={`select-country ${enabled ? 'active' : ''}`}
              onChange={(e) => {
                onChange(e.target.value);
                field.onChange(e);
              }}
            >
              <option value="">select country</option>
              {countries.map(({id, name}) => (
                <option key={id} value={id}>
                  {name}
                </option>
              ))}
            </select>
            {error && touched ? <div className="text-danger">{error}</div> : null}
          </>
        )}
      </Field>
    </div>
  );
};
CountrySelect.protoTypes = {
  name: PropTypes.string.isRequired,
  enabled: PropTypes.bool.isRequired,
  countries: PropTypes.arrayOf(PropTypes.shape({id: PropTypes.string.isRequired, name: PropTypes.string.isRequired})),
  error: PropTypes.string,
  touched: PropTypes.bool.isRequired
};
export default CountrySelect;
