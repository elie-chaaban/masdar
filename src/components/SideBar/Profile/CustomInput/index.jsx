import React from 'react';
import {Field} from 'formik';
import PropTypes from 'prop-types';

const CustomInput = ({className = 'col-6', type = 'text', name, enabled, title = null, error, touched}) => (
  <div className={`mt-2 ${className}`}>
    <p>{title ? title : name}</p>
    <Field
      className={`form-control ${enabled ? 'active' : ''}`}
      type={type}
      placeholder={title ? title : name}
      name={name}
      disabled={!enabled}
    />
    {error && touched ? <div className="text-danger">{error}</div> : null}
  </div>
);

CustomInput.protoTypes = {
  name: PropTypes.string.isRequired,
  enabled: PropTypes.bool.isRequired,
  title: PropTypes.string,
  className: PropTypes.string,
  type: PropTypes.string,
  error: PropTypes.string,
  touched: PropTypes.bool.isRequired
};

export default CustomInput;
