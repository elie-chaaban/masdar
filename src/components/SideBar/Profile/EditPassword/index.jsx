import React, {useState} from 'react';
import {Formik, Form, Field} from 'formik';
import passwordValidation from '../validations/password';
import {Col, Row} from 'react-bootstrap';
import {strengthColor, strengthIndicator} from './strength';
import {updatePassword} from '../../../../services/user';
import PropTypes from 'prop-types';
import {errorNotification, successNotification} from '../../../Utils/Notifications';

const EditPassword = ({click, profile}) => {
  const [indicator, setIndicator] = useState(undefined);
  const [loading, setLoading] = useState(false);
  const checkPassStrength = (value) => {
    const count = strengthIndicator(value);
    setIndicator(strengthColor(count));
  };
  const updatePass = async (password) => {
    try {
      setLoading(true);
      await updatePassword(profile, password);
      click();
      successNotification('Password Updated Successfully!');
    } catch (error) {
      errorNotification(error);
    } finally {
      setLoading(false);
    }
  };
  return (
    <Formik
      initialValues={{
        password: '',
        passwordConfirmation: ''
      }}
      validationSchema={passwordValidation}
      onSubmit={({password}) => {
        if (!loading) updatePass(password);
      }}
    >
      {({errors, touched, handleChange}) => {
        return (
          <Form className="password-form row">
            <div className="mt-4 col-6">
              <p>Password</p>
              <Field
                className="form-control"
                placeholder="password"
                name="password"
                type="password"
                onChange={(e) => {
                  handleChange(e);
                  checkPassStrength(e.target.value);
                }}
              />
              {errors.password && touched.password ? <div className="text-danger">{errors.password}</div> : null}
            </div>
            <div className=" mt-4  col-6 d-flex flex-column justify-content-center">
              <div className="password-strength">
                <p>Password strength</p>
                <div data-testid="indicator-bar" className={`bar ${indicator ? indicator.class : ''}`}></div>
                {indicator && <p data-testid="indicator-text">{indicator.label}</p>}
              </div>
            </div>
            <div className="mt-4 col-6">
              <p>Confirm Password</p>
              <Field className="form-control" placeholder="password Confirmation" name="passwordConfirmation" type="password" />
              {errors.passwordConfirmation && touched.passwordConfirmation ? (
                <div className="text-danger">{errors.passwordConfirmation}</div>
              ) : null}
            </div>
            <Col xs={12}>
              <Row>
                <Col xs={6}>
                  <button className="btn mt-3" type="submit">
                    {loading ? 'Saving...' : 'Save'}
                  </button>
                </Col>
              </Row>
            </Col>
          </Form>
        );
      }}
    </Formik>
  );
};
EditPassword.protoTypes = {
  click: PropTypes.func.isRequired
};
export default EditPassword;
