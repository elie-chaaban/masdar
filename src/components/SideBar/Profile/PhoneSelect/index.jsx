import React, {useMemo} from 'react';
import {getCountries, getCountryCallingCode} from 'react-phone-number-input/input';
import {Field} from 'formik';

const CountryCodeSelect = ({name, enabled}) => {
  const countryCode = useMemo(() => {
    let codes = [];
    getCountries().forEach((country) => {
      codes.push(getCountryCallingCode(country));
    });
    function onlyUnique(value, index, self) {
      return self.indexOf(value) === index;
    }
    return codes.filter(onlyUnique);
  }, []);
  return (
    <div className=" col-3 pr-1">
      <Field name={name}>
        {({
          field // { name, value, onChange, onBlur }
        }) => (
          <select {...field} className={`select-country ${enabled ? 'active' : ''}`} disabled={!enabled}>
            <option value="">code</option>
            {countryCode.map((code) => (
              <option key={code} value={code}>
                {code}
              </option>
            ))}
          </select>
        )}
      </Field>
    </div>
  );
};
export default CountryCodeSelect;
