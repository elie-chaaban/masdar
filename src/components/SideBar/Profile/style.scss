.profile {
  overflow-y: auto;
  position: absolute;
  top: 0;
  left: 45px;
  height: 100vh;
  text-transform: capitalize;
  z-index: 999;
  width: 98vw;
  background-color: rgba($color: #000000, $alpha: 0.9);
  p {
    color: white;
    margin-bottom: 0;
  }

  .btn {
    background-color: #314b77;
    color: white;
    border-radius: 0;
    padding: 10px 10px;
    text-transform: capitalize;
    width: 200px;
  }
  .header {
    background-color: #373737;
    .back {
      color: white;
      background-color: transparent !important;
      width: auto !important;
      padding: 0;
      box-shadow: none !important;
      &:hover {
        .arrow:nth-child(1) {
          transform: rotate(45deg);
          -webkit-transform: rotate(45deg);
        }
      }
      .arrow {
        border-style: solid;
        border-color: white;
        border-width: 0 0px 2px 2px;
        display: inline-block;
        vertical-align: middle;
        padding: 3px;
        transform: rotate(-135deg);
        -webkit-transform: rotate(-135deg);
        transition: transform 0.2s;
        &:nth-child(2) {
          transform: rotate(45deg);
          -webkit-transform: rotate(45deg);
        }
      }
    }
  }
  h1,
  h2,
  h3,
  h4,
  h5 {
    color: white;
    margin-bottom: 0;
  }
  .profile-pic {
    img {
      max-width: 120%;
    }
  }
  .user-info {
    .content {
      .form-control {
        border-radius: 0;
        padding: 0;
        margin: 0;
        width: 100%;
        box-shadow: none;
        transition: background 0.5s;
        outline: 0;
        padding: 0px 10px;
        height: 40px;
        font-size: 1rem;
        background-color: transparent !important;
        border: 0px;
        &.active {
          color: white;
          background-color: #373737 !important;
        }
      }
    }
  }
}
.password-form {
  .form-control {
    border-radius: 0;
    padding: 0;
    margin: 0;
    width: 100%;
    box-shadow: none;
    color: white;
    background-color: #373737 !important;
    outline: 0;
    height: 40px;
    padding: 0px 10px;
    border: 0px;
    font-size: 1rem;
  }
  .password-strength {
    .bar {
      width: 100%;
      position: relative;
      background-color: #373737 !important;
      height: 4px;
      overflow: hidden;
      &:after {
        position: absolute;
        top: 0;
        left: 0;
        height: 10px;
        transition: all 0.5s;
        content: '';
      }
      &.verylow:after {
        background-color: red;
        width: 25%;
      }
      &.low:after {
        background-color: yellow;
        width: 50%;
      }
      &.strong:after {
        background-color: orange;
        width: 75%;
      }
      &.verystrong:after {
        background-color: #1ed88f;
        width: 100%;
      }
    }
  }
}
.flag-select__btn:after {
  display: none !important;
}
.select-country {
  word-wrap: normal;
  width: 100%;
  padding: 0px 10px;
  background-color: transparent;
  color: gray;
  border: 0px;
  line-height: 1.5;
  height: 40px;
  font-size: 1rem;
  outline: 0;
  -webkit-appearance: none;
  -moz-appearance: none;
  &.active {
    color: white;
    background-color: #373737 !important;
  }
}

.select-city {
  word-wrap: normal;
  width: 100%;
  padding: 0px 10px;
  background-color: transparent;
  color: gray;
  border: 0px;
  line-height: 1.5;
  height: 40px;
  font-size: 1rem;
  outline: 0;
  -webkit-appearance: none;
  -moz-appearance: none;
  &.active {
    color: white;
    background-color: #373737 !important;
  }
}

@media screen and (max-width: 768px) {
  .profile {
    overflow-y: auto;
    position: absolute;
    top: 79px;
    left: 45px;
    height: 56vh;
    text-transform: capitalize;
    z-index: 999;
    width: 88vw;
    background-color: rgba(0, 0, 0, 0.9);
    p {
      color: white;
      margin-bottom: 0;
      font-size: 12px;
    }
    .btn {
      background-color: #314b77;
      color: white;
      border-radius: 0;
      padding: 10px 10px;
      text-transform: capitalize;
      width: 142px;
      font-size: 12px;
    }
    .header {
      height: 44px;
      background-color: #373737;
      .back {
        color: white;
        background-color: transparent !important;
        width: auto !important;
        padding: 0;
        box-shadow: none !important;
        &:hover {
          .arrow:nth-child(1) {
            transform: rotate(45deg);
            -webkit-transform: rotate(45deg);
          }
        }
        .arrow {
          border-style: solid;
          border-color: white;
          border-width: 0 0px 2px 2px;
          display: inline-block;
          vertical-align: middle;
          padding: 3px;
          transform: rotate(-135deg);
          -webkit-transform: rotate(-135deg);
          transition: transform 0.2s;
          &:nth-child(2) {
            transform: rotate(45deg);
            -webkit-transform: rotate(45deg);
          }
        }
      }
    }
    h1,
    h2,
    h3 {
      font-size: 12px;
      margin-top: 3px;
      margin-left: 1px;
    }
    h4,
    h5 {
      color: white;
      font-size: 19px;
      margin-top: 13px;
      margin-left: 15px;
    }
    .img-fluid {
      margin-top: 11px;
    }
    .profile-pic {
      img {
        max-width: 120%;
      }
    }
    .user-info {
      .content {
        .form-control {
          border-radius: 0;
          padding: 0;
          margin: 0;
          width: 100%;
          box-shadow: none;
          transition: background 0.5s;
          outline: 0;
          padding: 0px 10px;
          height: 40px;
          font-size: 12px;
          background-color: transparent !important;
          border: 0px;
          &.active {
            color: white;
            background-color: #373737 !important;
          }
        }
      }
    }
  }
  .select-country {
    word-wrap: normal;
    width: 100%;
    padding: 0px 10px;
    background-color: transparent;
    color: gray;
    border: 0px;
    line-height: 1.5;
    height: 40px;
    font-size: 12px;
    outline: 0;
    -webkit-appearance: none;
    -moz-appearance: none;
    &.active {
      color: white;
      background-color: #373737 !important;
    }
  }

  .select-city {
    word-wrap: normal;
    width: 100%;
    padding: 0px 10px;
    background-color: transparent;
    color: gray;
    border: 0px;
    line-height: 1.5;
    height: 40px;
    font-size: 12px;
    outline: 0;
    -webkit-appearance: none;
    -moz-appearance: none;
    &.active {
      color: white;
      background-color: #373737 !important;
    }
  }
  .password-form {
    .form-control {
      border-radius: 0;
      padding: 0;
      margin: 0;
      width: 100%;
      box-shadow: none;
      color: white;
      background-color: #373737 !important;
      outline: 0;
      height: 40px;
      padding: 0px 10px;
      border: 0px;
      font-size: 12px;
    }
    .password-strength {
      .bar {
        width: 100%;
        position: relative;
        background-color: #373737 !important;
        height: 4px;
        overflow: hidden;
        &:after {
          position: absolute;
          top: 0;
          left: 0;
          height: 10px;
          transition: all 0.5s;
          content: '';
        }
        &.verylow:after {
          background-color: red;
          width: 25%;
        }
        &.low:after {
          background-color: yellow;
          width: 50%;
        }
        &.strong:after {
          background-color: orange;
          width: 75%;
        }
        &.verystrong:after {
          background-color: #1ed88f;
          width: 100%;
        }
      }
    }
  }
}
