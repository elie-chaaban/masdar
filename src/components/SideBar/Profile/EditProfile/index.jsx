import React, {useState, useEffect, useCallback} from 'react';
import {Formik, Form, Field} from 'formik';
import CountrySelect from '../CountrySelect';
import CitySelect from '../CitySelect';
import CustomInput from '../CustomInput';
import CountryCodeSelect from '../PhoneSelect';
import userEditSchema from '../validations/userEdit';
import {updateProfileInfo} from '../../../../services/user';
import {errorNotification, successNotification} from '../../../Utils/Notifications';

const EditProfile = ({countries, profile}) => {
  const {firstName, mobileNumber, city, address1 = '', address2 = '', email, gender} = profile;

  const cityId = city ? city?.id : null;
  const countryId = city ? city?.country?.id : null;
  const countryCode = city ? city?.country?.code : null;

  const [edit, setEdit] = useState(false);
  const [loading, setLoading] = useState(false);
  const [countryCities, setCountryCities] = useState([]);
  useEffect(() => {
    if (countries && city && city.id) {
      const selectedCountry = countries.find((c) => c.id === city.country.id);
      setCountryCities(selectedCountry.cities);
    }
  }, [countries, city]);
  const onChangeCountry = useCallback(
    (countryId) => {
      const selectedCountry = countries.find((c) => c.id === countryId);
      if (selectedCountry) {
        setCountryCities(selectedCountry.cities);
      } else {
        setCountryCities([]);
      }
    },
    [countries]
  );
  const updateProfile = async (gender, mobileNumberCountryCode, mobileNumber, cityId, address1, address2) => {
    try {
      setLoading(true);
      await updateProfileInfo(profile, gender, mobileNumberCountryCode, mobileNumber, cityId, address1, address2);
      setEdit(false);
      successNotification('Profile Updated Successfully!');
    } catch (e) {
      errorNotification(e);
    } finally {
      setLoading(false);
    }
  };
  return (
    <Formik
      initialValues={{
        userName: firstName,
        email: email,
        phone: mobileNumber,
        phoneCode: countryCode,
        country: countryId,
        city: cityId,
        address1: address1,
        address2: address2
      }}
      validationSchema={userEditSchema}
      onSubmit={({phone, phoneCode, country, city, address1, address2}) => {
        if (!loading) updateProfile(gender, phoneCode, phone, city, address1, address2);
      }}
    >
      {({errors, touched, resetForm}) => (
        <Form className="content row">
          <CustomInput name="userName" enabled={false} />
          <CustomInput name="email" type="email" enabled={false} />
          <div className="mt-2 col-12">
            <p>Phone</p>
          </div>
          <div className="col-6">
            <div className="row">
              <CountryCodeSelect name="phoneCode" enabled={edit} />
              <div className="col-9 pl-1">
                <Field
                  className={`form-control ${edit ? 'active' : ''}`}
                  type="number"
                  placeholder="phone number"
                  name="phone"
                  disabled={!edit}
                />
              </div>
              <div className="col-12">
                {((errors.phone && touched.phone) || (errors.phoneCode && touched.phoneCode)) && (
                  <div className="text-danger">{errors.phone || errors.phoneCode}</div>
                )}
              </div>
            </div>
          </div>
          <div className="col-6" />
          <CountrySelect
            countries={countries}
            error={errors.country}
            touched={touched.country}
            name="country"
            enabled={edit}
            onChange={onChangeCountry}
          />
          <CitySelect cities={countryCities} error={errors.city} touched={touched.city} name="city" enabled={edit} />
          <CustomInput name="address1" error={errors.address1} touched={touched.address1} title="address 1" enabled={edit} />
          <CustomInput name="address2" title="address 2" enabled={edit} />
          <div className="col-12 mt-3">
            {edit && (
              <button className="btn mr-2" type="submit">
                {loading ? 'Saving...' : 'Save'}
              </button>
            )}
            <button
              className="btn"
              type="button"
              onClick={() => {
                if (edit) resetForm();
                setEdit(!edit);
              }}
            >
              {edit ? 'Cancel' : 'Edit'}
            </button>
          </div>
        </Form>
      )}
    </Formik>
  );
};

export default React.memo(EditProfile);
