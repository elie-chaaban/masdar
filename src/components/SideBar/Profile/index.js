import React, {useState, useEffect} from 'react';

import './style.scss';
import EditPassword from './EditPassword';
import profilepic from '../../../assets/images/profile.png';
import {Row, Col, Container} from 'react-bootstrap';
import {getProfileInfo} from '../../../services/user';
import EditProfile from './EditProfile';
import Loader from '../../Utils/Loader';
import useFetch from '../../../hooks';
import {Animate} from '../../Utils';

const Profile = ({active, click}) => {
  // eslint-disable-next-line no-unused-vars
  const [showEditPassword, setShowEditPassword] = useState(false);
  const {isLoading, response, fetch, isMountedRef} = useFetch(getProfileInfo);
  useEffect(() => {
    fetch();
    return () => (isMountedRef.current = false);
    // eslint-disable-next-line react-hooks/exhaustive-deps
  }, []);
  return (
    <Animate options={{duration: 500}} className="profile" unMountOnExit enter={active}>
      <div className="header py-2 px-4 d-flex justify-content-between">
        <h5>Profile</h5>
        <button className="btn back" onClick={click}>
          <i className="arrow"></i>
          <i className="arrow mr-2"></i>
        </button>
      </div>
      {isLoading ? (
        <div className="h-75 d-flex align-items-center justify-content-center">
          <Loader color="white">
            <p>Loading...</p>
          </Loader>
        </div>
      ) : (
        response && (
          <div className="content pt-4">
            <Container fluid>
              <Row className="px-4">
                <Col xs={1}>
                  <img src={profilepic} className="img-fluid" alt="profile" />
                </Col>
                <Col xs={11}>
                  <div className="user-info">
                    <div className="mb-2 content-header">
                      <Row className="border-bottom border-light pb-3 align-items-center">
                        <Col>
                          <h3>{response.info.firstName + ' ' + response.info.lastName}</h3>
                        </Col>
                        <Col xs={'auto'} className="ml-auto">
                          <button className="btn" onClick={() => setShowEditPassword(!showEditPassword)}>
                            {!showEditPassword ? 'Change Password' : 'Cancel'}
                          </button>
                        </Col>
                      </Row>
                      <div className="mt-4">
                        <p>{response.info.type === 'Admin' ? 'Administrator' : 'Not Administrator'}</p>
                      </div>
                    </div>
                    <Row>
                      <Col xs={6}>
                        <EditProfile countries={response.countries} profile={response.info} />
                      </Col>
                      <Col xs={6}>
                        <Animate enter={showEditPassword} unMountOnExit options={{duration: 500}}>
                          <EditPassword profile={response.info} click={() => setShowEditPassword(false)} />
                        </Animate>
                      </Col>
                    </Row>
                  </div>
                </Col>
              </Row>
            </Container>
          </div>
        )
      )}
    </Animate>
  );
};

export default Profile;
