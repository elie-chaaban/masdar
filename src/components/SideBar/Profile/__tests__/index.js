import React from 'react';
import {withHooks} from 'jest-react-hooks-shallow';
import {shallow} from 'enzyme';
import {act} from 'react-dom/test-utils';
import Profile from '..';
let resolve;
const profile = {info: {is_cms_user: 1, first_name: 'adel', last_name: 'chamas'}, countries: ['LB', 'AE']};
jest.mock('../../../../services/user', () => ({
  ...jest.requireActual('../../../../services/user'),
  getProfileInfo: () =>
    new Promise((res) => {
      resolve = res;
    })
}));

test('Active Profile Component', () => {
  withHooks(async () => {
    const click = jest.fn();
    let dom = shallow(<Profile active={true} click={click} />);
    dom.update();
    expect(dom.find('.profile')).toHaveLength(1);
    expect(dom).toMatchSnapshot();
    dom.find('.back').simulate('click');
    expect(click).toBeCalledTimes(1);
    expect(dom.find('Loader')).toHaveLength(1);
    expect(dom.find('Animate').prop('enter')).toBeTruthy();
    expect(dom.find('.content')).toHaveLength(0);
    await act(async () => resolve(profile));
    dom.update();
    expect(dom.find('Loader')).toHaveLength(0);
    expect(dom.find('.content')).toHaveLength(1);
    expect(dom.find('h3').text()).toBe(profile.info.first_name + ' ' + profile.info.last_name);
    expect(dom.find('button').at(1).text()).toBe('Change Password');
    expect(dom.find('.mt-4 p').text()).toBe('Administrator');
    expect(dom.find('Memo(EditProfile)').prop('countries')).toBe(profile.countries);
    expect(dom.find('Memo(EditProfile)').prop('profile')).toBe(profile.info);
    expect(dom.find('Animate').at(1).prop('enter')).toBeFalsy();
    expect(dom.find('EditPassword')).toHaveLength(1);
    await act(async () => {
      dom.find('button').at(1).simulate('click');
    });
    dom.update();
    expect(dom.find('Animate').at(1).prop('enter')).toBeTruthy();
    await act(async () => {
      dom.find('EditPassword').props().click();
    });
    dom.update();
    expect(dom.find('Animate').at(1).prop('enter')).toBeFalsy();
    jest.clearAllMocks();
  });
});
