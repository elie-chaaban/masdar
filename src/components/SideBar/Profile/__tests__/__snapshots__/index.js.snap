// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Active Profile Component 1`] = `
<Animate
  animation="fade"
  className="profile"
  enter={true}
  onAnimationEnd={null}
  options={
    Object {
      "duration": 500,
    }
  }
  unMountOnExit={true}
>
  <div
    className="header py-2 px-4 d-flex justify-content-between"
  >
    <h5>
      Profile
    </h5>
    <button
      className="btn back"
      onClick={[MockFunction]}
    >
      <i
        className="arrow"
      />
      <i
        className="arrow mr-2"
      />
    </button>
  </div>
  <div
    className="h-75 d-flex align-items-center justify-content-center"
  >
    <Loader
      color="white"
    >
      <p>
        Loading...
      </p>
    </Loader>
  </div>
</Animate>
`;
